sdk/python/kfp/_client.py

```diff
@@ -906,7 +901,7 @@ def __init__(self, spec, resource_references):
             pipeline_json_string = json.dumps(pipeline_obj)
         api_params = [
             kfp_server_api.ApiParameter(
-                name=sanitize_k8s_name(name=k, allow_capital_underscore=True),
```

Comment:
This seems to be a behavior change? Do we understand the impact, whether it's acceptable?